Dockerfile

```diff
@@ -21,72 +24,29 @@ SHELL ["/bin/bash", "-o", "pipefail", "-c"]
-  apt-get -qq dist-upgrade && \
-  echo "applying workaround for ubuntu/postfix bug described in https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuecomment-855326403" && \
-  mv /bin/hostname{,.bak} && \
-  echo "echo docker-mailserver.invalid" > /bin/hostname && \
-  chmod +x /bin/hostname && \
-  apt-get -qq install postfix && \
-  mv /bin/hostname{.bak,} && \
-  apt-get -qq --no-install-recommends install \
-  # A - D
-  altermime amavisd-new apt-transport-https arj binutils bzip2 bsd-mailx \
-  ca-certificates cabextract clamav clamav-daemon cpio curl \
-  dbconfig-no-thanks dovecot-core dovecot-fts-xapian dovecot-imapd \
-  dovecot-ldap dovecot-lmtpd dovecot-managesieved dovecot-pop3d \
-  dovecot-sieve dovecot-solr dumb-init \
-  # E - O
-  ed fetchmail file gamin gnupg gzip iproute2 \
-  locales logwatch lhasa libdate-manip-perl libldap-common liblz4-tool \
-  libmail-spf-perl libnet-dns-perl libsasl2-modules lrzip lzop \
-  netcat-openbsd nftables nomarch opendkim opendkim-tools opendmarc \
-  # P - Z
-  pax pflogsumm postgrey p7zip-full postfix-ldap postfix-pcre \
-  postfix-policyd-spf-python postsrsd pyzor \
-  razor rpm2cpio rsyslog sasl2-bin spamassassin supervisor \
-  unrar-free unzip uuid whois xz-utils && \
-  # Fail2Ban
-  gpg --keyserver ${FAIL2BAN_GPG_PUBLIC_KEY_SERVER} \
-    --recv-keys ${FAIL2BAN_GPG_PUBLIC_KEY_ID} 2>&1 && \
-  curl -Lkso fail2ban.deb ${FAIL2BAN_DEB_URL} && \
-  curl -Lkso fail2ban.deb.asc ${FAIL2BAN_DEB_ASC_URL} && \
-  FINGERPRINT=$(LANG=C gpg --verify \
-  fail2ban.deb.asc fail2ban.deb 2>&1 \
-    | sed -n 's#Primary key fingerprint: \(.*\)#\1#p') && \
-  if [[ -z ${FINGERPRINT} ]]; then \
-    echo "ERROR: Invalid GPG signature!" >&2; exit 1; fi && \
-  if [[ ${FINGERPRINT} != "${FAIL2BAN_GPG_FINGERPRINT}" ]]; then \
-    echo "ERROR: Wrong GPG fingerprint!" >&2; exit 1; fi && \
-  dpkg -i fail2ban.deb 2>&1 && \
-  rm fail2ban.deb fail2ban.deb.asc && \
-  # cleanup
-  apt-get -qq autoremove && \
-  apt-get -qq autoclean && \
-  apt-get -qq clean && \
-  rm -rf /var/lib/apt/lists/* && \
-  c_rehash 2>&1
-
+COPY target/scripts/build/* /build/
 COPY target/scripts/helpers/log.sh /usr/local/bin/helpers/log.sh
-COPY target/bin/sedfile /usr/local/bin/sedfile
+RUN /bin/bash /build/packages.sh
 
+COPY target/bin/sedfile /usr/local/bin/sedfile
 RUN chmod +x /usr/local/bin/sedfile
 
+
 # -----------------------------------------------
 # --- ClamAV & FeshClam -------------------------
 # -----------------------------------------------
 
 RUN \
   echo '0 */6 * * * clamav /usr/bin/freshclam --quiet' >/etc/cron.d/clamav-freshclam && \
   chmod 644 /etc/clamav/freshclam.conf && \
-  freshclam && \
   sedfile -i 's/Foreground false/Foreground true/g' /etc/clamav/clamd.conf && \
   mkdir /var/run/clamav && \
   chown -R clamav:root /var/run/clamav && \
   rm -rf /var/log/clamav/
 
+# hadolint ignore=DL3021
+COPY --link --from=docker.io/clamav/clamav:latest /var/lib/clamav /var/lib/clamav
```

Comment:
Hmm, I am not sure. I think the command is pretty self-explanatory, but on the other hand, I don't mind either. While the rate-limiting indeed was a problem, I think listing problems we solve in comments in the whole code base would increase code base size by 9000% 😆

Reply:
> I think listing problems we solve in comments in the whole code base would increase code base size by 9000%

I don't like it either sometimes, but history is messy with all the refactoring, and I've seen some past contributions make assumptions because they weren't aware of why something was done :\

When refactoring settles down, we can probably offload plenty of it, especially if we introduce some technical maintainer doc pages.

For now a small comment is much easier to get context about a decision than wading through PRs I review verbosely :stuck_out_tongue: